optax/_src/complex_valued_test.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2019 DeepMind Technologies Limited. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for `complex_valued.py`."""
+
+from absl.testing import absltest
+
+import chex
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+from optax._src import alias
+from optax._src import complex_valued
+from optax._src import update
+
+
+def _loss_fun_complex_to_real(z):
+  return (z.conj() * z).real.sum()
+
+
+def _loss_fun_real_to_real(params):
+  x, y = params
+  return _loss_fun_complex_to_real(x + y * 1j)
+
+
+class ComplexValuedTest(chex.TestCase):
+
+  @chex.all_variants
+  def test_split_complex(self):
+
+    def do_update(loss_fun, optimizer, params, opt_state):
+      loss, grads = jax.value_and_grad(loss_fun)(params)
+      # Complex gradients need to be conjugated before being added to parameters
```

Comment:
It's a choice of convention on the complex gradients in JAX, which is different from PyTorch and other packages. It's discussed in [this section](https://gist.github.com/wdphy16/118aef6fb5f82c49790d7678cf87da29#conjugate-of-complex-gradients) of the proposal.

Reply:
Sounds good! Actually, @wdphy16 would you be able to expand the docstring in the new complex module to explain how the complex number support works (based on the split real norm section of your proposal) and to mention the conjugating (based on the conjugate of complex gradients section of your proposal with the links you have there). I think just two paragraphs would help a lot and it would just be a slight rewording of what you have in the proposal already.